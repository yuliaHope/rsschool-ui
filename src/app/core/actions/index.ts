export * from './schedule';
export * from './user';
export * from './courses';
export * from './assignments';
