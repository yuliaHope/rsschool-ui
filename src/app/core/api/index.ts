export * from './schedule';
export * from './session';
export * from './user';
export * from './courses';
export * from './assignments';
