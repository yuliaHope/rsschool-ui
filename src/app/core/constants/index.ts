export * from './schedule';
export * from './courses';
export * from './user';
export * from './assignments';
