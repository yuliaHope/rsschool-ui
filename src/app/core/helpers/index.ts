export * from './schedule';
export * from './assignment';
