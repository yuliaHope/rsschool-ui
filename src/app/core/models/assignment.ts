export enum AssignmentStatus {
    Assigned = 'Assigned',
    ReadyForReview = 'ReadyForReview',
    Rejected = 'Rejected',
    Checked = 'Checked',
}

export interface IAssignment {
    taskId: number;
    courseId: string;
    studentId: string;
    mentorId: string;
    studentComment: string;
    mentorComment: string;
    score: number;
    assignmentRepo: string;
    deadlineDate: number;
    completeDate: number;
    checkDate: number;
    status: AssignmentStatus;
    [key: number]: any;
}

export interface IAssignmentDocument extends IAssignment {
    _id: string;
}
