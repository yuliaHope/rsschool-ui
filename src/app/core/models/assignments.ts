export enum AssignmentStatus {
    Assigned = 'Assigned',
    Checked = 'Checked',
    Rejected = 'Rejected',
    ReadyForReview = 'ReadyForReview',
    MissedDeadline = 'MissedDeadline',
}

export interface IAssignment {
    taskId: string;
    title: string;
    courseId: string;
    studentId: string;
    mentorId: string;
    studentComment: string;
    mentorComment: string;
    score: number;
    assignmentRepo: string;
    deadlineDate: number;
    completeDate: number;
    checkDate: number;
    status: AssignmentStatus;
}

export interface IAssignmentModel extends IAssignment {
    _id: string;
}
