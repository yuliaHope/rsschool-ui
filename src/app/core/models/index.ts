export * from './events';
export * from './stage';
export * from './user';
export * from './course';
export * from './assignments';
export * from './courseMentor';
export * from './courseStudent';
