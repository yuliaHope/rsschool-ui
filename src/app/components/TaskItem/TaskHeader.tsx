import * as React from 'react';
import { classNames } from '../../core/styles';
import { AssignmentStatus } from '../../core/models';
const cn = classNames(require('./index.scss'));

type Props = {
    status: string;
    score: number;
};
export const TaskHeader = ({ status, score }: Props) => {
    let headerTxt;
    let mark;
    let headerClassNames;
    if (status === AssignmentStatus.Assigned) {
        headerTxt = 'Not submitted yet!';
        mark = '';
        headerClassNames = 'bg-warning';
    } else if (status === AssignmentStatus.MissedDeadline) {
        headerTxt = 'The deadline has passed!';
        mark = '';
        headerClassNames = 'bg-danger';
    } else if (status === AssignmentStatus.Checked) {
        headerTxt = 'Done!';
        mark = score;
        headerClassNames = mark >= 100 ? 'bg-success' : 'bg-warning';
    } else {
        headerTxt = status;
        mark = '';
        headerClassNames = '';
    }

    const colorizeText = headerClassNames ? 'text-white' : '';

    return (
        <div className={`card-header ${headerClassNames} ${colorizeText} ${cn('header-alignment')}`}>
            <span className="card-header-text"> {headerTxt} </span>
            {mark ? <span className="card-header-text">{mark}%</span> : null}
        </div>
    );
};
