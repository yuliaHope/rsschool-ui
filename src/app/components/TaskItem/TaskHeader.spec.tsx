import * as React from 'react';
import { shallow } from 'enzyme';
import { TaskHeader } from './TaskHeader';
import { AssignmentStatus } from '../../core/models';

describe('Task Item', () => {
    const props = {
        status: AssignmentStatus.Assigned,
        score: 0,
    };
    describe('render Task Header', () => {
        it('in case status is Assigned', () => {
            const output = shallow(<TaskHeader {...props} />);
            expect(output).toMatchSnapshot();
        });
        it('in case status is MissedDeadline', () => {
            const nextProps = {
                ...props,
                status: AssignmentStatus.MissedDeadline,
            };
            const output = shallow(<TaskHeader {...nextProps} />);
            expect(output).toMatchSnapshot();
        });
        it('in case status is Checked && score >= 100', () => {
            const nextProps = {
                status: AssignmentStatus.Checked,
                score: 100,
            };
            const output = shallow(<TaskHeader {...nextProps} />);
            expect(output).toMatchSnapshot();
        });
        it('in case status is Checked && score < 100', () => {
            const nextProps = {
                status: AssignmentStatus.Checked,
                score: 99,
            };
            const output = shallow(<TaskHeader {...nextProps} />);
            expect(output).toMatchSnapshot();
        });
        it('in case status is ReadyForReview', () => {
            const nextProps = {
                ...props,
                status: AssignmentStatus.ReadyForReview,
            };
            const output = shallow(<TaskHeader {...nextProps} />);
            expect(output).toMatchSnapshot();
        });
        it('in case status is Rejected', () => {
            const nextProps = {
                ...props,
                status: AssignmentStatus.Rejected,
            };
            const output = shallow(<TaskHeader {...nextProps} />);
            expect(output).toMatchSnapshot();
        });
    });
});
