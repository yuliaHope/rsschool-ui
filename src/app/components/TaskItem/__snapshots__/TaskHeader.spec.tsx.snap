// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Item render Task Header in case status is Assigned 1`] = `
<div
  className="card-header bg-warning text-white header-alignment"
>
  <span
    className="card-header-text"
  >
     
    Not submitted yet!
     
  </span>
</div>
`;

exports[`Task Item render Task Header in case status is Checked && score < 100 1`] = `
<div
  className="card-header bg-warning text-white header-alignment"
>
  <span
    className="card-header-text"
  >
     
    Done!
     
  </span>
  <span
    className="card-header-text"
  >
    99
    %
  </span>
</div>
`;

exports[`Task Item render Task Header in case status is Checked && score >= 100 1`] = `
<div
  className="card-header bg-success text-white header-alignment"
>
  <span
    className="card-header-text"
  >
     
    Done!
     
  </span>
  <span
    className="card-header-text"
  >
    100
    %
  </span>
</div>
`;

exports[`Task Item render Task Header in case status is MissedDeadline 1`] = `
<div
  className="card-header bg-danger text-white header-alignment"
>
  <span
    className="card-header-text"
  >
     
    The deadline has passed!
     
  </span>
</div>
`;

exports[`Task Item render Task Header in case status is ReadyForReview 1`] = `
<div
  className="card-header   header-alignment"
>
  <span
    className="card-header-text"
  >
     
    ReadyForReview
     
  </span>
</div>
`;

exports[`Task Item render Task Header in case status is Rejected 1`] = `
<div
  className="card-header   header-alignment"
>
  <span
    className="card-header-text"
  >
     
    Rejected
     
  </span>
</div>
`;
